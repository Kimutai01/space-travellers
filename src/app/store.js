import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from '../redux/missions/missions';
import rocketReducer from '../redux/rocket/rocket';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
  },
});

export default store;
