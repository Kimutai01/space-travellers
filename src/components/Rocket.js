import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchedRockets, updateRockets } from '../redux/rocket/rocket';
import './Rocket.scss';

const Rocket = () => {
  const dispatch = useDispatch();
  const allRockets = useSelector((state) => state.rockets);
  useEffect(() => {
    if (!allRockets.length) dispatch(fetchedRockets());
  }, [dispatch]);

  const handleReserve = ({ target }) => {
    const { id } = target;
    dispatch(updateRockets(Number(id)));
  };

  return (
    <div>
      {allRockets.map((rocket) => (
        <div className="section" key={rocket.id}>
          <img className="rocket-image" src={rocket.flickr_image} alt="rocket" />
          <div className="description">
            <h2>{rocket.rocket_name}</h2>
            <p>
              <span>
                {rocket.canceled && <button className="btn btn-secondary" type="button">Reserved</button>}
              </span>
              {rocket.description}
            </p>
            {!rocket.canceled && (
              <button id={rocket.id} onClick={handleReserve} className="btn btn-primary mybtn" type="button">
                Reserve
              </button>
            )}
            {rocket.canceled && (
              <button id={rocket.id} onClick={handleReserve} className="btn btn-danger mybtn" type="button">
                Cancel
              </button>
            )}
          </div>
        </div>
      ))}
    </div>
  );
};

export default Rocket;
