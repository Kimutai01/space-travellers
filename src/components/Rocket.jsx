import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import rocket, { fetchedRockets, updateRockets } from '../redux/rocket/rocket';

const Rocket = () => {
  const dispatch = useDispatch();
  const allRockets = useSelector((state) => state.rockets);
  useEffect(() => {
    if (!allRockets.length) dispatch(fetchedRockets());
  }, [dispatch]);

  const handleReserve = ({ target }) => {
    const { id } = target;
    dispatch(updateRockets(Number(id)));
  };

  return (
    <div>
      {allRockets.map((rocket) => (
        <div key={rocket.id}>
          <div className="rocket-image">
            <img src={rocket.flickr_image} alt="rocket image" />
          </div>

          <h2>{rocket.rocket_name}</h2>
          <p>
            <span>
              {rocket.canceled && <button type="button">Reserved</button>}
            </span>
            {rocket.description}
          </p>
          {!rocket.canceled && (
            <button id={rocket.id} onClick={handleReserve} type="button">
              Reserve
            </button>
          )}
          {rocket.canceled && (
            <button id={rocket.id} onClick={handleReserve} type="button">
              Cancel
            </button>
          )}
        </div>
      ))}
    </div>
  );
};

export default Rocket;
