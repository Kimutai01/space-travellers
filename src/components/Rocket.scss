.rocket-image {
  width: 300px;
  padding: 18px;
}

.section {
  display: flex;

  .description {
    display: flex;
    flex-direction: column;

    h2 {
      font-size: 20px;
    }
  }

  .mybtn {
    width: 80px;
  }
}
