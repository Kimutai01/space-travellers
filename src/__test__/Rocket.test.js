import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import Rocket from '../components/Rocket';
import store from '../app/store';

test('Rocket component testing', () => {
  const TREE = TestRenderer.create(
    <Provider store={store}>
      <Rocket />
    </Provider>,
  );
  expect(TREE).toMatchSnapshot();
});
