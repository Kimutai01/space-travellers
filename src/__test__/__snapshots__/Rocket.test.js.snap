// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket component testing 1`] = `<div />`;
